File: GeoFirestoreExample/geofirestore/src/main/java/org/imperiumlabs/geofirestore/AndroidEventRaiser.java
package org.imperiumlabs.geofirestore;

import android.os.Handler;
import android.os.Looper;

// FULLY TESTED

class AndroidEventRaiser implements EventRaiser {
    private final Handler mainThreadHandler;

    public AndroidEventRaiser() {
        this.mainThreadHandler = new Handler(Looper.getMainLooper());
    }

    @Override
    public void raiseEvent(Runnable r) {
        this.mainThreadHandler.post(r);
    }
}

File: geofirestore/src/main/java/org/imperiumlabs/geofirestore/AndroidEventRaiser.kt
package org.imperiumlabs.geofirestore

import android.os.Handler
import android.os.Looper

// FULLY TESTED

class AndroidEventRaiser: EventRaiser {
    private val mainThreadHandler = Handler(Looper.getMainLooper())

    override fun raiseEvent(r: Runnable) {
        this.mainThreadHandler.post(r)
    }
}
